pkg/models/registries/image_test.go

```diff
@@ -0,0 +1,82 @@
+package registries
+
+import (
+	"testing"
+)
+
+func TestParseImage(t *testing.T) {
+	type image struct {
+		ImageName string
+		ExDomain  string
+		ExTag     string
+		ExPath    string
+	}
+
+	testImages := []image{
+		{ImageName: "dockerhub.qingcloud.com/kubesphere/test:v1", ExDomain: "dockerhub.qingcloud.com", ExTag: "v1", ExPath: "kubesphere/test"},
+		{ImageName: "harbor.devops.kubesphere.local:30280/library/tomcat:latest", ExDomain: "harbor.devops.kubesphere.local:30280", ExTag: "latest", ExPath: "library/tomcat"},
+		{ImageName: "zhuxiaoyang/nginx:v1", ExDomain: "docker.io", ExTag: "v1", ExPath: "zhuxiaoyang/nginx"},
+		{ImageName: "nginx", ExDomain: "docker.io", ExTag: "latest", ExPath: "library/nginx"},
+	}
+
+	for _, image := range testImages {
+		res, err := ParseImage(image.ImageName)
+		if err != nil {
+			t.Fatalf("Get err %s", err)
+		}
+
+		if res.Domain != image.ExDomain {
+			t.Fatalf("Doamin got %v, expected %v", res.Domain, image.ExDomain)
+		}
+
+		if res.Tag != image.ExTag {
+			t.Fatalf("Tag got %v, expected %v", res.Tag, image.ExTag)
+		}
+
+		if res.Path != image.ExPath {
+			t.Fatalf("Path got %v, expected %v", res.Path, image.ExPath)
+		}
+	}
+
+	invalidImage := []image{
+		{ImageName: "http://docker.io/nginx:latest"},
+		{ImageName: "https://harbor.devops.kubesphere.local:30280/library/tomcat:latest"},
+		{ImageName: "docker.io/nginx:latest:latest"},
+		{ImageName: "nginx:8000:latest"},
+	}
+
+	for _, image := range invalidImage {
+		_, err := ParseImage(image.ImageName)
+		if err == nil {
+			t.Fatalf("Parse invalid image but without get any error")
+		}
+	}
+
+}
+
+func TestStringWithoutScheme(t *testing.T) {
+	type testRawurl struct {
+		Rawurl string
+		ExUrl  string
+	}
+	testRawurls := []testRawurl{
+		{"http://dockerhub.qingcloud.com/kubesphere/nginx:v1", "dockerhub.qingcloud.com/kubesphere/nginx:v1"},
+		{"https://dockerhub.qingcloud.com/kubesphere/nginx:v1", "dockerhub.qingcloud.com/kubesphere/nginx:v1"},
+		{"dockerhub.qingcloud.com/kubesphere/nginx:v1", "dockerhub.qingcloud.com/kubesphere/nginx:v1"},
```

Comment:
add test case like 'nginx:latest'

Reply:
why marked this?

Reply:
only an image name start with "http" will be used this func.